core/src/test/java/org/apache/accumulo/core/data/KeyExtentTest.java

```diff
@@ -181,4 +182,42 @@ private KeyExtent writeAndReadFields(KeyExtent in) throws IOException {
     return KeyExtent.readFrom(new DataInputStream(bais));
   }
 
+  @Test
+  public void testClip() {
+    // Test extent will null end/prevEnd and infinite range matches original
+    ke = new KeyExtent(TableId.of("1"), null, null);
+    Range range = new Range();
+    KeyExtent expected = ke;
+    KeyExtent clipped = ke.clip(range);
+    assertEquals(expected, clipped);
+
+    // Test extent with null end/prevEnd with a range, should match the range
+    expected = new KeyExtent(TableId.of("1"), new Text("r_0001"), new Text("r_0000"));
+    ke = new KeyExtent(TableId.of("1"), null, null);
+    range = new Range("r_0000", false, "r_0001", true);
+    clipped = ke.clip(range);
+    assertEquals(expected, clipped);
+
+    // Test resulting extent is narrowed, the overlapping rows are 4 - 7
+    expected = new KeyExtent(TableId.of("1"), new Text("r_0007"), new Text("r_0003"));
+    ke = new KeyExtent(TableId.of("1"), new Text("r_0007"), new Text("r_0000"));
+    range = new Range("r_0003", false, "r_0009", true);
+    clipped = ke.clip(range);
+    assertEquals(expected, clipped);
+
+    // Test disjoint
+    ke = new KeyExtent(TableId.of("1"), new Text("r_0007"), new Text("r_0005"));
+    range = new Range("r_0002", false, "r_0004", true);
+    clipped = ke.clip(range, true);
+    assertNull(clipped);
+    final Range r = range;
+    assertThrows(IllegalArgumentException.class, () -> ke.clip(r));
+
+    // Invalid row ranges
+    assertThrows(IllegalArgumentException.class,
+        () -> ke.clip(new Range("r_0002", true, "r_0004", true)));
+    assertThrows(IllegalArgumentException.class, () -> ke
+        .clip(new Range(new Key("r_0002", "cf_0002"), false, new Key("r_0003", "cf_0003"), true)));
```

Comment:
Could add an additional test where the start row is empty string, this would exercise the code in isExclusiveKey that checks the key length.